app/uk/gov/hmrc/tariffclassificationfrontend/controllers/StatusChangeAction.scala

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import play.api.data.Form
+import play.api.mvc.{Action, AnyContent, Call, Request}
+import play.twirl.api.Html
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, Operator}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+trait StatusChangeAction[T] extends RenderCaseAction {
+
+  protected val authenticatedAction: AuthenticatedAction
+
+  protected val form: Form[T]
+
+  protected def status(c: Case): T
+
+  protected def chooseStatusView(c: Case, preFilledForm: Form[T])(implicit request: Request[_]): Html
+
+  protected def update(c: Case, status: T, operator: Operator)(implicit hc: HeaderCarrier): Future[Case]
+
+  protected def onSuccessRedirect(reference: String): Call
+
+  def chooseStatus(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+    getCaseAndRenderView(
+      reference,
+      c => successful(chooseStatusView(c, form.fill(status(c))))
+    )
+  }
+
+  def updateStatus(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+    form.bindFromRequest().fold(
+      errors => {
+        getCaseAndRenderView(
+          reference,
+          c => successful(chooseStatusView(c, errors))
+        )
+      },
+      (status: T) => {
+        getCaseAndRespond(
+          reference,
+          c => {
+            if(statusHasChanged(c, status)) {
```

Comment:
not about the refactor. But maybe move this logic into the service?

Reply:
Potentially I guess the question is: Is checking if a status has changed business logic, or view layer?

I don't have a strong opinion either way